common/types/provider.go

```diff
@@ -442,6 +445,133 @@ func nativeToValue(a ref.TypeAdapter, value interface{}) (ref.Val, bool) {
 	return nil, false
 }
 
+// RefValueToExprValue converts between ref.Val and exprpb.ExprValue.
+func RefValueToExprValue(res ref.Val, err error) (*exprpb.ExprValue, error) {
```

Comment:
Keep in mind that we want the conversions to be symmetric, so you'll want the counterparts to these methods which are still present in the `server.go` file.

For the sake of simplicity consider only translating back and forth from `exprpb.Value` objects and leave the other methods in the server.

Reply:
So we should just move ref.Val <-> exprpb.ExprValue to provider.go riight?

Reply:
Only the `ref.Val` <-> `Value` functions for now, but not the `ExprValue` related functions at the moment.